test/unit/com/unifina/datasource/DataSourceEventQueueSpec.groovy

```diff
@@ -147,4 +146,43 @@ class DataSourceEventQueueSpec extends Specification {
 		})
 	}
 
+	void "timer drift"() {
+		DataSourceEventQueue queue = createQueue()
+		int eventsProcessed = 0
+
+		when:
+		Thread consumerThread = Thread.start {
+			queue.start()
+		}
+
+		Thread producerThread = Thread.start {
+			queue.enqueue(new Event<Integer>(1, new Date(10000), new Consumer<Integer>() {
+				@Override
+				void accept(Integer integer) {
+					eventsProcessed++
+				}
+			}))
+			queue.enqueue(new Event<Integer>(2, new Date(20000), new Consumer<Integer>() {
```

Comment:
Add slightly more testing power to not have an even second as the end event timestamp

```suggestion
			queue.enqueue(new Event<Integer>(2, new Date(20100), new Consumer<Integer>() {
```